### PR TITLE
Defines the d.ts multi-factor related comments for reference generation.

### PR DESCRIPTION
**Pull request by bojeil-google**
Adds missing multi-factor related claims to the `DecodedIdToken` interface.

### CODE REVIEW

File: src/index.d.ts [hunk 1]
```diff
@@ -544,6 +544,48 @@ declare namespace admin.auth {
     toJSON(): Object;
   }
 
+  /**
+   * Interface representing the common properties of a user enrolled second factor.
+   */
+  interface MultiFactorInfo {
+  
+    /**
+     * The unique identifier of the enrolled second factor.
```
Comment (hiranya911) [inline src/index.d.ts]: Nit: This is the user's unique ID from the 2nd factor right? Perhaps make that a bit clear.

Comment (bojeil-google) [reply src/index.d.ts]: Done.

File: src/index.d.ts [hunk 2]
```diff
@@ -544,6 +544,48 @@ declare namespace admin.auth {
     toJSON(): Object;
   }
 
+  /**
+   * Interface representing the common properties of a user enrolled second factor.
+   */
+  interface MultiFactorInfo {
+  
+    /**
+     * The user's unique ID of the enrolled second factor.
```
Comment (kevinthecheung) [inline src/index.d.ts]: The ID of the enrolled second factor. This ID is unique to the user.

File: src/index.d.ts [hunk 3]
```diff
@@ -544,6 +544,48 @@ declare namespace admin.auth {
...
+     */
+    uid: string;
+
+    /**
+     * The optional display name for an enrolled second factor.
```
Comment (kevinthecheung) [inline src/index.d.ts]: The optional display name of the enrolled second factor.

File: src/index.d.ts [hunk 4]
```diff
@@ -544,6 +544,48 @@ declare namespace admin.auth {
...
+     */
+    displayName?: string;
+
+    /**
+     * The optional date the second factor was enrolled, formatted as a UTC string.
+     */
+    enrollmentTime?: string;
+
+    /**
+     * The second factor type identifier. For SMS second factor, this is `"phone"`.
```
Comment (kevinthecheung) [inline src/index.d.ts]: The type identifier of the second factor. For SMS second factors, this is `phone`.

File: src/index.d.ts [hunk 5]
```diff
@@ -639,14 +681,21 @@ declare namespace admin.auth {
      */
     tenantId?: string | null;
 
+    /**
+     * The multi-factor related properties for the current user if available.
```
Comment (kevinthecheung) [inline src/index.d.ts]: user, if

File: src/index.d.ts [hunk 6]
```diff
@@ -639,14 +681,21 @@ declare namespace admin.auth {
...
-      enrolledFactors: Array<{
-        uid: string;
-        phoneNumber: string;
-        displayName?: string;
-        enrollmentTime?: string;
-        factorId: string;
-      }>;
+
+      /**
+       * List of enrolled second factors on the current user record.
```
Comment (kevinthecheung) [inline src/index.d.ts]: List of second factors enrolled with the current user.

File: src/index.d.ts [hunk 7]
```diff
@@ -696,17 +785,49 @@ declare namespace admin.auth {
...
-      enrolledFactors: Array<{
-        uid?: string;
-        phoneNumber: string;
-        displayName?: string;
-        enrollmentTime?: string;
-        factorId: string;
-      }> | null;
+
+      /**
+       * The user's updated list of enrolled second factors. The provided list will always
```
Comment (kevinthecheung) [inline src/index.d.ts]: The updated list of enrolled second factors. The provided list overwrites the user's existing list of second factors. When null is passed, all of the user's existing second factors are removed.

File: src/index.d.ts [hunk 8]
```diff
@@ -791,6 +915,19 @@ declare namespace admin.auth {
        */
       sign_in_provider: string;
 
+      /**
+       * This is the type identifier or `factorId` of the second factor, provided the
```
Comment (kevinthecheung) [inline src/index.d.ts]: The type identifier..

File: src/index.d.ts [hunk 9]
```diff
@@ -791,6 +915,19 @@ declare namespace admin.auth {
...
+       * ID token was obtained from a multi-factor authenticated user.
+       * For phone, this is `“phone”`.
+       */
+      sign_in_second_factor?: string;
+
+      /**
+       * This is the `uid` of the second factor used to sign in, provided the
```
Comment (kevinthecheung) [inline src/index.d.ts]: The `uid` of..

File: src/index.d.ts (hunk 3 above)
Comment (bojeil-google) [reply src/index.d.ts]: Done.

File: src/index.d.ts (hunk 4 above)
Comment (bojeil-google) [reply src/index.d.ts]: Done.

File: src/index.d.ts (hunk 5 above)
Comment (bojeil-google) [reply src/index.d.ts]: Done.

File: src/index.d.ts (hunk 6 above)
Comment (bojeil-google) [reply src/index.d.ts]: Done.

File: src/index.d.ts (hunk 7 above)
Comment (bojeil-google) [reply src/index.d.ts]: Done.

File: src/index.d.ts (hunk 8 above)
Comment (bojeil-google) [reply src/index.d.ts]: Done.

File: src/index.d.ts (hunk 9 above)
Comment (bojeil-google) [reply src/index.d.ts]: Done.

File: src/index.d.ts (hunk 2 above)
Comment (bojeil-google) [reply src/index.d.ts]: Done.

**Pull request merged**